webapp/channels/src/utils/utils.tsx

```diff
@@ -212,6 +212,48 @@ const removeQuerystringOrHash = (extin: string): string => {
 };
 
 export const getFileType = (extin: string): typeof FileTypes[keyof typeof FileTypes] => {
+    // Handle null or undefined input
+    if (!extin) {
+        return FileTypes.OTHER;
+    }
+
+    // Special handling for image proxy URLs
+    // Check for various forms of image proxy URLs
+    if (extin.includes('/api/v4/image') &&
+        (extin.includes('?url=') || extin.includes('&url='))) {
+        return FileTypes.IMAGE;
+    }
+
+    // Check for image file extensions in the URL path
+    try {
+        // Try to parse as a URL - this will validate if it's a proper URL
+        const url = new URL(extin);
+        const pathname = url.pathname;
+        const pathParts = pathname.split('/');
+        const lastPathPart = pathParts[pathParts.length - 1];
+
+        if (lastPathPart && lastPathPart.includes('.')) {
+            const urlExtension = lastPathPart.split('.').pop()?.toLowerCase();
+            if (urlExtension && Constants.IMAGE_TYPES.indexOf(urlExtension) > -1) {
+                return FileTypes.IMAGE;
+            }
+        }
+    } catch (e) {
+        // If it's not a valid URL, try a simpler approach
+        // This handles cases where extin might be a file path but not a valid URL
+        const urlPath = extin.split('?')[0]; // Remove query parameters
+        const pathParts = urlPath.split('/');
+        const lastPathPart = pathParts[pathParts.length - 1];
+
+        if (lastPathPart && lastPathPart.includes('.')) {
+            const urlExtension = lastPathPart.split('.').pop()?.toLowerCase();
+            if (urlExtension && Constants.IMAGE_TYPES.indexOf(urlExtension) > -1) {
+                return FileTypes.IMAGE;
+            }
+        }
```

Comment:
confused: if it's not a valid URL, why try to parse it "like a URL" at all? Can we just ignore the exception on parse and continue on to try to match purely on extension?

Reply:
ok, I think I went too far in terms of defensive programming here. Ok, what I can do is still be a little bit deffensive without being too aggresive-defensive :D , so will have the logic reduced to: If it's a valid URL, parse it properly and 
If it's not a valid URL, just check if the string itself has an extension and that it is all. My next commit will have this.